src/content/guides/creating-your-own-admission-controller/index.md

```diff
@@ -0,0 +1,219 @@
+
+In the second, there is a **mutation admission controller** which modifies the resource submitted by the user, so you can make some defaults or force some schema. The cluster admins can attach mutation webhooks to the API to be run in the same way as validation. Indeed mutation logic runs one step before the validation.
+
+__Note:__ I would like to mention that Kubernetes API allows you to register custom resource objects too, called `Custom Resource Definition`. In addition, you can create an API extension server which listens to a new REST path. But in this guide, we focus exclusively on the admission functionality.
+
+## Goal
+
+Our goal here is to create a simple validation controller which will empower us to influence the pod creation. Although there are many more possibilities and the logic can be as complex as you need. The goal is to create a basic version which makes a simple validation. You can find more real examples in the links at the bottom.
+
+Our example controller will be called `grumpy` and will reject all new pods with a name different than `smooth-app`. I recognize it may be tempting to deploy this controller in a real cluster ;).
+
+## How API proxies requests
+
+The Kubernetes API server needs to know when to send the incoming request to our admission controller. The Kubernetes nature advocates the use of a declarative state always and this is no exception. Below we define a `ValidationWebhookConfiguration` which gives the needed information to the API:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: grumpy
+webhooks:
+  - name: grumpy.giantswarm.io
+    clientConfig:
+      service:
+        name: grumpy
+        namespace: default
+        path: "/validate"
+      caBundle: "${CA_BUNDLE}"
+    rules:
+      - operations: ["CREATE"]
+        apiGroups: [""]
+        apiVersions: ["v1"]
+        resources: ["pods"]
+```
+
+The configuration definition is pretty obvious. You can see there are two main parts to be considered. In the first one, `clientConfig`, we define where our service can be found (can be an external URL too), and the `path` which our validation server will listen. Also, you notice there is a `CA` to be defined. Since security is always important, adding the cert authority will tell Kubernetes API to use HTTPS and validate our server using the passed asset. In the next section, you will find an explanation on how to generate all the certs needed.
```

Comment:
```suggestion
You can see there are two main parts to be considered. In the first one, `clientConfig`, we define where our service can be found (this can be an external URL too), and the `path` which our validation server will listen on. Also, you notice there is a `CA` to be defined. Since security is always important, adding the cert authority will tell Kubernetes API to use HTTPS and validate our server using the passed asset. In the next section, you will find an explanation on how to generate all the certs needed.
```

try to not state things "matter-of-factly" to some ppl it might not be obvious and they'd be discouraged to go on with this.